cmd/signer/Dockerfile

```diff
@@ -0,0 +1,34 @@
+# syntax=docker/dockerfile:1.6
+
+FROM scratch as creds
+
+FROM --platform=${BUILDPLATFORM} mcr.microsoft.com/oss/go/microsoft/golang:1.21@sha256:f5fc8bf2873d2fa538b2d87e6e8f9bc6d05a3c419a840147b25f6386855192e9 AS signer-builder
+WORKDIR /build
+COPY . .
+ENV CGO_ENABLED=0
+ARG TARGETARCH TARGETOS GOFLAGS=-trimpath
+
+ENV GOOS=${TARGETOS} GOARCH=${TARGETARCH} GOFLAGS=${GOFLAGS}
+RUN \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    go build -o /signer ./cmd/signer
+
+FROM mcr.microsoft.com/azure-cli:cbl-mariner2.0 as az-base
+WORKDIR /root
+COPY ./tmp/xsignextension-0.46-py3-none-any.whl .
```

Comment:
Obviously this won't work. We'll have to get the extension from a keyvault using secrets for this build..

Reply:
the extension itself doesn't need to be in a key vault does it? We could fetch this from storage?

Reply:
Yeah, I meant "get the credentials from a keyvault to access the file from storage"

Reply:
Irrelevant since this will be moved elsewhere.